### PR TITLE
Harden stream keep-alive ACK tracking

### PR DESCRIPTION
**Pull request by PhilippeFerreiraDeSousa**
## Summary

- replace the single pending keep-alive ACK slot with a bounded ordered window on Android and iOS
- accept delayed ACKs for the active stream, retire stale older requests, and reset the missed-ACK streak
- report a sustained missed-ACK threshold once while continuing heartbeat monitoring instead of automatically stopping the stream
- document the non-destructive timeout behavior

## Tests

- `./scripts/check-android-compile.sh bluetooth-sdk`
- `cd mobile/android && ./gradlew --no-daemon --console=plain -PmentraPublicSdk=true :mentra-bluetooth-sdk:testDebugUnitTest`
- `cd mobile/android && ./gradlew --no-daemon --console=plain -PmentraPublicSdk=true :mentra-bluetooth-sdk:testDebugUnitTest --tests com.mentra.bluetoothsdk.StreamKeepAliveAckWindowTest`
- `cd mobile/modules/bluetooth-sdk && xcodebuild -quiet -scheme MentraBluetoothSDK -destination "platform=iOS Simulator,id=4B226A23-64AA-4D31-BE6D-C100E0118FBF" test CODE_SIGNING_ALLOWED=NO`

## Validation note

- Native policy and build paths are covered on both platforms; no live glasses streaming session was available for hardware validation.

---

> [!NOTE]
> **Medium Risk**
> Changes live streaming reliability semantics (no auto-stop on keep-alive failure) and BLE timing behavior on both native SDKs; apps that relied on implicit stop may need to react to the new error event explicitly.
> 
> **Overview**
> **Stream keep-alive handling** on Android and iOS no longer tracks a single pending ACK. Both platforms use a shared **`StreamKeepAliveAckWindow`** that keeps a bounded, ordered list of recent `ackId`s so delayed ACKs still count, stale IDs drop out when a newer ACK arrives, and the miss streak resets on a valid match.
> 
> **Timeout behavior is non-destructive:** after three missed intervals the SDK still emits a **`stream_status` error** (message notes that monitoring continues), but it **does not** tear down the keep-alive monitor or call **`stopStream()`**. Heartbeats keep going so a late ACK can recover the session. The README documents this contract for integrators.
> 
> Monitors created after a successful **`startStream`** use **`forStartedStream`**, which arms the window immediately so the first heartbeat can participate in miss detection; non-terminal stream statuses still call **`arm()`** on first transition for externally managed paths.
> 
> Unit tests on Kotlin and Swift cover late ACKs, one-shot timeout reporting, and bounded-window edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8d07a7399d538f7adf5640742173574c1fa9f0a2. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: mobile/modules/bluetooth-sdk/ios/Source/MentraBluetoothSDK.swift [hunk 1]
```diff
@@ -1557,28 +1555,10 @@ public final class MentraBluetoothSDK {
...
-                        resolvedConfig: nil
-                    )
-                )
-                delegate?.mentraBluetoothSDK(self, didReceive: .streamStatus(event))
-                stopStreamKeepAliveMonitor()
-                DeviceManager.shared.stopStream()
-                return
-            }
-        }
+        let missedAckThreshold = tracker.ackWindow.recordTick()
```
[the remark this answers is not in the document]
Comment (PhilippeFerreiraDeSousa) [reply mobile/modules/bluetooth-sdk/ios/Source/MentraBluetoothSDK.swift:1557]: Fixed cross-platform in 6e28fd8d6c. Managed stream monitors are created only after startStream receives a successful status, so their ACK windows now use the armed forStartedStream construction path before the first keep-alive is sent. Externally managed streams still do not create an SDK monitor. Added Android and iOS regression tests proving the first request reaches the missed-ACK threshold; focused Android tests and the 15-test iOS package suite pass.

File: mobile/modules/bluetooth-sdk/ios/Source/streaming/StreamKeepAliveAckWindow.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+/// Tracks a bounded, ordered window of stream keep-alive requests.
+///
+/// ACKs can arrive after a newer request has already been sent. Matching any
+/// request still in the window proves that the stream transport is responsive;
+/// requests older than the match can then be discarded as stale.
+final class StreamKeepAliveAckWindow {
+    private let maxTrackedAckIds: Int
+    private let maxMissedAcks: Int
+    private var pendingAckIds: [String] = []
```
[the remark this answers is not in the document]
Comment (PhilippeFerreiraDeSousa) [reply mobile/modules/bluetooth-sdk/ios/Source/streaming/StreamKeepAliveAckWindow.swift:11]: Fixed cross-platform in 8d07a7399d. Managed monitors now retain maxMissedAcks + 1 request IDs, so the oldest request remains matchable when the threshold tick sends the next heartbeat. Added Android and iOS regressions for an ACK arriving three intervals late; the Android suite passes 7 tests and the iOS package passes 16 tests.

**Comment by PhilippeFerreiraDeSousa** [comment]
Superseded by #3987 (OS-1937). Dedicated heartbeat ACK repair is no longer the streaming ownership design; the replacement removes the obsolete watchdog and covers native controller liveness, publisher failure reporting, and stale callbacks together. This branch should not be merged separately.
